### PR TITLE
Kernel allocator

### PR DESCRIPTION
**Pull request by mnurczynski**
Finished the kernel allocator (finally)!

Functions availible:

kset(uint32_t size, void* offset)
kmalloc(uint32_t size)
kfree(void* pointer)
krealloc(void* pointer, uint32_t new_size)
kmres(void* pointer, uint32_t size)

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -78,12 +78,12 @@ clean:
 
 # Invoke QEMU wihtout waiting for GDB
 run: build/final.iso
-	qemu-system-i386 -monitor stdio -cdrom ./build/final.iso -boot a
+	qemu-system-i386 -singlestep -monitor stdio -cdrom ./build/final.iso -boot a
```
Comment (MateuszTk) [inline makefile]: Remove `-singlestep`

**Review by MateuszTk — APPROVED** [review]
ok

File: makefile (hunk 1 above)
Comment (mnurczynski) [reply makefile]: Will be removed during conflict resolution, as it does not warrant a separate commit.

File: src/kernel/kmalloc.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @return None.
+ */
+extern void kfree(void* pointer);
+
+
+/**
+ * @brief This function reserves (permamently takes out of the allocating pool) the given memory area at specified address.
+ *
+ * @note This one is quite slow too. It's recomended to run it right after kset on system startum.
+ *       If you run in on an already allocated area - the behaviour is undefined (so run it before using kmalloc).
```
Comment (magistermaks) [inline src/kernel/kmalloc.h:44]: can `kmres()` calls overlap (partially or entirely)?

**Review by magistermaks — APPROVED** [review]

**Pull request merged**